test/test_support.ts

```diff
@@ -85,15 +86,15 @@ export interface GoldenFileTest {
 }
 
 export function goldenTests(): GoldenFileTest[] {
-  var tsExtRe = /\.ts$/;
-  var testFolder = path.join(__dirname, '..', '..', 'test_files');
-  var files = fs.readdirSync(testFolder).filter((fn) => !!fn.match(tsExtRe));
-  return files.map((fn) => {
+  var testInputGlob = path.join(__dirname, '..', '..', 'test_files', '**', '*.in.ts');
+  var testInputs = glob.sync(testInputGlob);
+  return testInputs.map((testPath) => {
+    let testName = testPath.match(/\/test_files\/(.*)\.in\.ts$/)[1];
     return {
-      name: fn,
-      tsPath: path.join(testFolder, fn),
-      sicklePath: path.join(testFolder, 'sickle', fn),
-      es6Path: path.join(testFolder, 'es6', fn.replace(tsExtRe, '.js')),
+      name: testName,
+      tsPath: testPath,
+      sicklePath: testPath.replace(/\.in\.ts$/, '.sickle.ts'),
+      es6Path: testPath.replace(/\.in\.ts$/, '.tr.js'),
```

Comment:
translated, and I thought the entire word is a bit too long. `out` sorts before `sickle`. I could rename `sickle` to `middle` or `intermediate` or so. I really don't think it matters too much as long as we have the files close to each other.